_appmap/env.py

```diff
@@ -133,6 +134,29 @@ def display_params(self):
     def getLogger(self, name) -> trace_logger.TraceLogger:
         return cast(trace_logger.TraceLogger, logging.getLogger(name))
 
+    def determine_log_file(self):
+        log_file = "appmap.log"
+
+        # Try creating the log file in the current directory
+        try:
+            with open(log_file, 'a', encoding='UTF8'):
+                pass
+        except IOError:
```

Comment:
I don't think we want to create the log file in the user's home directory, or as a temp file.

Instead, let's just write a message to stderr return `None`. The circumstances in which creation is going to fail are also those in which the user doesn't care whether there's a log file (e.g. when starting a REPL).